src/messaging/messaging-types.ts

```diff
@@ -789,44 +775,49 @@ function validateLightSettings(lightSettings: LightSettings) {
   }
-
-  if (typeof lightSettings.lightOffDurationMillis !== 'undefined') {
-    if (!validator.isNumber(lightSettings.lightOffDurationMillis) || lightSettings.lightOffDurationMillis < 0) {
-      throw new FirebaseMessagingError(
-        MessagingClientErrorCode.INVALID_PAYLOAD,
-        'android.notification.lightSettings.lightOffDurationMillis must be a non-negative duration in milliseconds');
-    }
-    const duration: string = transformMillisecondsToSecondsString(lightSettings.lightOffDurationMillis);
-    (lightSettings as any).lightOffDurationMillis = duration;
+  if (!validator.isNumber(lightSettings.lightOnDurationMillis) || lightSettings.lightOnDurationMillis < 0) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD,
+      'android.notification.lightSettings.lightOnDurationMillis must be a non-negative duration in milliseconds');
   }
+  const durationOn: string = transformMillisecondsToSecondsString(lightSettings.lightOnDurationMillis);
+  (lightSettings as any).lightOnDurationMillis = durationOn;
 
-  if (typeof lightSettings.color !== 'undefined' && validator.isString(lightSettings.color)) {
-    if (!/^#[0-9a-fA-F]{6}$/.test(lightSettings.color)) {
-      throw new FirebaseMessagingError(
-        MessagingClientErrorCode.INVALID_PAYLOAD,
-        'android.notification.lightSettings.color must be in the form #RRGGBB or a `Color` object');
-    }
-    const rgb = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(lightSettings.color);
-    const color: Color = {
-      red: parseInt(rgb[1], 16) / 255.0,
-      green: parseInt(rgb[2], 16) / 255.0,
-      blue: parseInt(rgb[3], 16) / 255.0,
-      alpha: 1.0,
-    };
-    (lightSettings as any).color = color;
+  if (typeof lightSettings.lightOffDurationMillis === 'undefined' || lightSettings.lightOffDurationMillis === null) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD,
+      'android.notification.lightSettings.lightOffDurationMillis is required');
   }
+  if (!validator.isNumber(lightSettings.lightOffDurationMillis) || lightSettings.lightOffDurationMillis < 0) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD,
+      'android.notification.lightSettings.lightOffDurationMillis must be a non-negative duration in milliseconds');
+  }
+  const durationOff: string = transformMillisecondsToSecondsString(lightSettings.lightOffDurationMillis);
+  (lightSettings as any).lightOffDurationMillis = durationOff;
 
-  validateColor(lightSettings.color as Color);
+  if (typeof lightSettings.color === 'undefined' || lightSettings.color === null) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD, 'android.notification.lightSettings.color is required');
+  }
+  if (!/^#[0-9a-fA-F]{6}$/.test(lightSettings.color) && !/^#[0-9a-fA-F]{8}$/.test(lightSettings.color)) {
```

Comment:
Lets add `!validator.isString(lightSettings.color) || ....` which should cover undefined and null cases.